src/vcCamera.cpp

```diff
@@ -45,46 +89,8 @@ void vcCamera_UpdateSmoothing(vcState * pProgramState, vcCamera *pCamera, vcCame
       udDouble3 step = pCamInput->smoothTranslation * udMin(1.0, stepAmount * sCameraTranslationSmoothingSpeed);
       pCamera->position += step;
       pCamInput->smoothTranslation -= step;
-      if (pProgramState->settings.camera.mapMode[pProgramState->activeViewportIndex])
-      {
-        for (int v = 0; v < vcMaxViewportCount; v++)
-        {
-          if (!pProgramState->settings.camera.mapMode[v])
-          {
-            if (pProgramState->geozone.projection != udGZPT_ECEF)
-            {
-                  pProgramState->pViewports[v].camera.position.x = pCamera->position.x;
-                  pProgramState->pViewports[v].camera.position.y = pCamera->position.y;
-            }
-            else
-            {
-              udDouble3 camMapPosLatLong = udGeoZone_CartesianToLatLong(pProgramState->geozone, pCamera->position);
-              udDouble3 camPosLatLong = udGeoZone_CartesianToLatLong(pProgramState->geozone, pProgramState->pViewports[v].camera.position);
-              camMapPosLatLong.z = camPosLatLong.z;
-              pProgramState->pViewports[v].camera.position = udGeoZone_LatLongToCartesian(pProgramState->geozone, camMapPosLatLong);
-            }
-          }
-        }
-      }
-      else
-      {
-        for (int v = 0; v < vcMaxViewportCount; v++)
-        {
-          if (pProgramState->settings.camera.mapMode[v])
-          {
-            if (pProgramState->geozone.projection != udGZPT_ECEF)
-              pProgramState->pViewports[v].camera.position = udDouble3::create(pCamera->position.x, pCamera->position.y, pProgramState->pViewports[v].camera.position.z);
-            else
-            {
-              udDouble3 camPosLatLong = udGeoZone_CartesianToLatLong(pProgramState->geozone, pCamera->position);
-              udDouble3 camMapPosLatLong = udGeoZone_CartesianToLatLong(pProgramState->geozone, pProgramState->pViewports[v].camera.position);
-              camPosLatLong.z = camMapPosLatLong.z;
-              pProgramState->pViewports[v].camera.position = udGeoZone_LatLongToCartesian(pProgramState->geozone, camPosLatLong);
-            }
-            pProgramState->pViewports[v].camera.headingPitch = udDouble2::create(0.0f, -UD_HALF_PI);
-          }
-        }
-      }
+
+      vcCamera_UpdateCameraMap(pProgramState, pCamera);
```

Comment:
It seems like all of the changes in this file could have been resolved just be changing the if above so this block gets called whenever there is a mapMode regardless of `if (udMagSq3(pCamInput->smoothTranslation) > minSmoothingThreshold)`?

Reply:
Fixed, now just call vcCamera_UpdateCameraMap for every movement of the camera.